src/ngraph_builder.cc

```diff
@@ -41,6 +41,66 @@ const static std::map<tf::DataType, ngraph::element::Type> TF_NGRAPH_TYPE_MAP =
      {tf::DataType::DT_UINT16, ng::element::u16},
      {tf::DataType::DT_BOOL, ng::element::boolean}};
 
+template<size_t a, size_t b, size_t c, size_t d>
+static void Reshape(std::shared_ptr<ng::Node>& ng_node) {
+  static_assert(a < 4 && b < 4 && c < 4 && d < 4,
+                "Number of dimensions cannot exceed 4");
+  static_assert(a != b && a != c && a != d && b !=c && b != d && c != d,
+                "Dimensions indices cannot be equal");
+  auto& s = ng_node->get_shape();
+  ng::Shape reshaped_shape{s[a], s[b], s[c], s[d]};
+  NGRAPH_VLOG(3) << "reshaped_shape: " << ng::join(reshaped_shape);
+  ng_node = make_shared<ng::op::Reshape>(
+      ng_node, ng::AxisVector{a, b, c, d}, reshaped_shape);
+}
+
+static void NhwcToNgraph() {}
+
+template <typename... Arguments, typename T>
+static void NhwcToNgraph(const std::vector<T>& source,
+                         std::vector<size_t>& dest,
+                         Arguments&&... remaining) {
+  dest[0] = source[1];
+  dest[1] = source[2];
+  NhwcToNgraph(remaining...);
+}
+template <typename... Arguments>
+static void NhwcToNgraph(std::shared_ptr<ng::Node>& ng_node,
```

Comment:
While Variadic templates are cool, not sure why we need it here. The main advantage of a Variadic template is to be able to cascade computations - but in this case the last stage is a no-op (i.e., function in line 57). Please remove the variadic part. 

Then add a few C++ unit tests with various corner cases.

Reply:
Maybe I'm missing something but that's how variadic functions work -- you need a terminal condition.

I'll add unit tests though.

Reply:
Yes - I am not confused about how the variadic templates work - my point is that I don't see the need for it in this use case.

Reply:
What would it look like otherwise, then.